tensorflow_addons/optimizers/discriminative_layer_training_test.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for Discriminative Layer Training Manager for TensorFlow."""
+
+import tensorflow as tf
+from tensorflow_addons.utils import test_utils
+import numpy as np
+from tensorflow_addons.optimizers.discriminative_layer_training import (
+    DiscriminativeWrapper,
+)
+import itertools
+import os
+from tensorflow.python.eager import context
+
+
+def toy_cnn():
+    """Consistently create model with same random weights
+    skip head activation to allow both bce with logits and cce with logits
+
+    The model returned by this function should have identical weights to all
+    other models returned by this function, for the duration of that
+    continuous integration run
+
+    model is intended to work with
+    x = np.ones(shape = (None, 32, 32, 3), dtype = np.float32)
+    y = np.zeros(shape = (None, 5), dtype = np.float32)
+    y[:, 0] = 1.
+    """
+
+    cnn_model_path = "cnn.h5"
+
+    if not os.path.exists(cnn_model_path):
+        # force eager mode for simple initialization of vars
+        with context.eager_mode():
+            tf.random.set_seed(1)
+            bignet = tf.keras.applications.mobilenet_v2.MobileNetV2(
+                include_top=False, weights=None, input_shape=(32, 32, 3), pooling="avg"
+            )
+
+            # take the first few layers so we cover BN, Conv, Pooling ops for testing
+            net = tf.keras.models.Model(
+                inputs=bignet.input, outputs=bignet.get_layer("block_2_add").output
+            )
+            model = tf.keras.Sequential(
+                [
+                    net,
+                    tf.keras.layers.GlobalAveragePooling2D(),
+                    tf.keras.layers.Dropout(0.5),
+                    tf.keras.layers.Dense(5, name="head"),
+                ]
+            )
+            # always save and never return initialized model from memory
+            # it seems you cannot pass variables from a nested eager context to its parent graph context
+
+            model.save(cnn_model_path)
```

Comment:
Slightly confused by this. Does this save the h5 in a temp directory or in project root if the tests are ran?

Reply:
I'm actually not too sure about that one. I think it saves it in the project root. I could look into having it save in a temp directory?